stdlib/public/core/Slice.swift

```diff
@@ -223,7 +223,11 @@ extension Slice: Collection {
     try _base.withContiguousStorageIfAvailable { buffer in
       let start = _base.distance(from: _base.startIndex, to: _startIndex)
       let count = _base.distance(from: _startIndex, to: _endIndex)
-      let slice = UnsafeBufferPointer(rebasing: buffer[start ..< start + count])
+      let (end, overflow) = start.addingReportingOverflow(count)
```

Comment:
Is it possible to get the codegen or performance characteristics you want by using operations that resemble index advancement rather than integer arithmetic?